tests/abi/functions.h

```diff
@@ -0,0 +1,309 @@
+#pragma once
+
+/*
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+
+#include <assert.h>
+#include <stdint.h>
+#include <stdio.h>
+
+#include "for_each.h"
+
+/*
+ * Helper macros
+ */
+
+#ifndef ABIDEF
+#define ABIDEF
+#endif
+
+#define STRINGIFY_HELPER(X) #X
+#define STRINGIFY(X) STRINGIFY_HELPER(X)
+
+#define CONCATENATE_HELPER(LHS, RHS) LHS##RHS
+#define CONCATENATE(LHS, RHS) CONCATENATE_HELPER(LHS, RHS)
+
+#define CONCATENATE_THREE_HELPER(F, S, T) F##S##T
+#define CONCATENATE_THREE(F, S, T) CONCATENATE_THREE_HELPER(F, S, T)
+
+#define DO_NOTHING_HELPER(LHS, RHS) LHS RHS
+#define DO_NOTHING(LHS, RHS) DO_NOTHING_HELPER(LHS, RHS)
+
+static_assert(__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+                || __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__,
+              "Unsupported endianness (pdp? something crazy? Memory "
+              "corruption?)");
+
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+#define PRINT_FUNCTION_HEADER(FUNCTION_NAME)            \
+  puts("  Function: \"" STRINGIFY(FUNCTION_NAME) "\"\n" \
+       "  IsLittleEndian: yes");
```

Comment:
Why print endianess really relevant?

Reply:
This helps making the artifact self-sufficient. As in, the parsing side don't need to be aware of the system endianness.

Reply:
Go-o-o-one!